src/components/RadioButton/RadioButton.tsx

```diff
@@ -0,0 +1,50 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import React from 'react';
+import { ComponentPropsWithoutRef } from 'react';
+
+export interface RadioButtonProps extends ComponentPropsWithoutRef<'div'> {
+  radioId: string;
+}
+
+export function RadioButton(props: RadioButtonProps) {
+  let { radioId, ...rest } = props;
+
+  return (
+    <div data-testid="radio-button">
+      <div className="inline-flex items-center">
+        <label className="relative flex cursor-pointer items-center rounded-full p-3">
+          <input
+            id={radioId}
+            name={radioId}
+            type="radio"
+            className={`before:content[''] peer relative h-5 w-5
+                        cursor-pointer appearance-none rounded-full
+                        border-2 border-gray-400
+                        before:left-2/4 before:block before:h-12
+                        before:w-12 before:-translate-y-2/4
+                        before:-translate-x-2/4 before:rounded-full
+                        before:bg-blue-gray-500 before:opacity-0
+                        before:transition-opacity before:absolute before:top-2/4
+                        checked:before:bg-green-500 transition-all
+                        checked:border-green-500 hover:before:opacity-10`}
+            {...rest}
```

Comment:
A bit of challenge: I did quite one with much less classes: https://play.tailwindcss.com/PjkqZ5Jhcx

Is that all needed!? 🤔

Reply:
yes, it's more complicated than the instructions... Sorry...
I changed:

```ts
className={`peer relative h-5 w-5
                        cursor-pointer appearance-none rounded-full
                        border-2 border-gray-400 checked:border-green-500`}
            {...rest}
```

Reply:
Thanks for that!